File: 수진/Programmers/Greedy/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        
        //무게 순으로 정렬
        Arrays.sort(people);
        
        int smallIdx = 0; //최소 몸무게 사람의 인덱스       
        int bigIdx = people.length - 1; //최대 몸무게 사람의 인덱스
        
        //아직 안 탄 사람들 중, 두 사람 몸무게의 합이 limit를 넘는지 확인
        while(smallIdx <= bigIdx){
            if(people[smallIdx] + people[bigIdx] <= limit){
                smallIdx ++;
            }
            bigIdx --;
            answer ++;
        }
        
        return answer;
    }
}

File: 나영/May/구명보트.cpp
#include <string>
#include <vector>
#include<algorithm>
#include<functional>
#include<iostream>
#include<cstdio>
using namespace std;
// 최대 두명 무게 제한 있음

int solution(vector<int> people, int limit) {
    int answer = 0;
    sort(people.begin(), people.end());

    int left = 0;
    int right = people.size()-1;
    while(left<=right){
    	if(people[left]+people[right]<=limit){
    		left++;
		}
		right--;
		answer++;
    	
	}
    return answer;
}
int main(){
	vector<int> people{20,70,80,30};
	int limit = 100;
	int answer = solution(people,limit);
	cout<<answer;
	
}

